File: src/Cxx/Utilities/BrownianPoints.cxx
#include <vtkSmartPointer.h>

#include <vtkActor.h>
#include <vtkArrowSource.h>
#include <vtkBrownianPoints.h>
#include <vtkCamera.h>
#include <vtkGlyph3D.h>
#include <vtkMath.h>
#include <vtkPolyData.h>
#include <vtkPolyDataMapper.h>
#include <vtkProperty.h>
#include <vtkRenderWindow.h>
#include <vtkRenderWindowInteractor.h>
#include <vtkRenderer.h>
#include <vtkSphereSource.h>

#include <vtkNamedColors.h>

int main(int, char *[])
{
  // Create a sphere
  vtkSmartPointer<vtkSphereSource> sphereSource =
    vtkSmartPointer<vtkSphereSource>::New();
  
  // Generate random vectors
  vtkMath::RandomSeed(5070); // for testing
  vtkSmartPointer<vtkBrownianPoints> brownianPoints =
    vtkSmartPointer<vtkBrownianPoints>::New();
  brownianPoints->SetInputConnection(sphereSource->GetOutputPort());
  
  vtkSmartPointer<vtkArrowSource> arrowSource =
    vtkSmartPointer<vtkArrowSource>::New();
  
  vtkSmartPointer<vtkGlyph3D> glyph3D =
    vtkSmartPointer<vtkGlyph3D>::New();
  glyph3D->SetSourceConnection(arrowSource->GetOutputPort());
  glyph3D->SetInputConnection(brownianPoints->GetOutputPort());
  glyph3D->SetScaleFactor(.3);
  
  // Create a mapper and actor for sphere
  vtkSmartPointer<vtkPolyDataMapper> mapper = 
    vtkSmartPointer<vtkPolyDataMapper>::New();
  mapper->SetInputConnection(sphereSource->GetOutputPort());
  
  vtkSmartPointer<vtkActor> actor = 
    vtkSmartPointer<vtkActor>::New();
  actor->GetProperty()->EdgeVisibilityOn();
  actor->GetProperty()->SetInterpolationToFlat();
  actor->SetMapper(mapper);

  // Create a mapper and actor for glyphs
  vtkSmartPointer<vtkPolyDataMapper> glyphMapper =
    vtkSmartPointer<vtkPolyDataMapper>::New();
  glyphMapper->SetInputConnection(glyph3D->GetOutputPort());

  vtkSmartPointer<vtkNamedColors> colors =
    vtkSmartPointer<vtkNamedColors>::New();

  vtkSmartPointer<vtkActor> glyphActor =
    vtkSmartPointer<vtkActor>::New();
  glyphActor->GetProperty()->SetColor(colors->GetColor3d("banana").GetData());
  glyphActor->SetMapper(glyphMapper);

  //Create a renderer, render window, and interactor
  vtkSmartPointer<vtkRenderer> renderer = 
    vtkSmartPointer<vtkRenderer>::New();
  vtkSmartPointer<vtkRenderWindow> renderWindow = 
    vtkSmartPointer<vtkRenderWindow>::New();
  renderWindow->AddRenderer(renderer);
  vtkSmartPointer<vtkRenderWindowInteractor> renderWindowInteractor = 
    vtkSmartPointer<vtkRenderWindowInteractor>::New();
  renderWindowInteractor->SetRenderWindow(renderWindow);

  //Add the actor to the scene
  renderer->AddActor(actor);
  renderer->AddActor(glyphActor);

  // Create a nice view
  renderer->ResetCamera();
  renderer->GetActiveCamera()->Azimuth(30);
  renderer->GetActiveCamera()->Elevation(30);
  renderer->GetActiveCamera()->Dolly(1.4);
  renderer->ResetCameraClippingRange();

  renderer->SetBackground(colors->GetColor3d("SlateGray").GetData());
  
  //Render and interact
  renderWindow->SetSize(640, 480);
  renderWindow->Render();
  renderWindowInteractor->Start();

  return EXIT_SUCCESS;
}

File: src/Java/Utilities/BrownianPoints.java
import vtk.vtkNativeLibrary;
import vtk.vtkRenderWindow;
import vtk.vtkRenderWindowInteractor;
import vtk.vtkRenderer;
import vtk.vtkActor;
import vtk.vtkNamedColors;
import vtk.vtkPolyDataMapper;
import vtk.vtkSphereSource;
import vtk.vtkArrowSource;
import vtk.vtkBrownianPoints;
import vtk.vtkGlyph3D;
import vtk.vtkMath;

public class BrownianPoints 
{
  // -----------------------------------------------------------------
  // Load VTK library and print which library was not properly loaded
  static 
  {
    if (!vtkNativeLibrary.LoadAllNativeLibraries()) 
    {
      for (vtkNativeLibrary lib : vtkNativeLibrary.values()) 
      {
        if (!lib.IsLoaded()) 
        {
          System.out.println(lib.GetLibraryName() + " not loaded");
        }
      }
    }
    vtkNativeLibrary.DisableOutputWindow(null);
  }
  // -----------------------------------------------------------------

  public static void main(String args[]) 
  {
    vtkNamedColors Color = new vtkNamedColors(); 

    //For Actor Color
    double glyphActorColor[] = new double[4];
    //Renderer Background Color
    double BgColor[] = new double[4];
         
    //Change Color Name to Use your own Color for Change Actor Color
    Color.GetColor("Banana",glyphActorColor);
    //Change Color Name to Use your own Color for Renderer Background
    Color.GetColor("SlateGray",BgColor);
    
    //Create a sphere
    vtkSphereSource sphereSource = new vtkSphereSource();

    //Generate random vectors
    vtkMath math = new vtkMath(	);
    math.RandomSeed(5070); // for testing
    vtkBrownianPoints brownianPoints = new vtkBrownianPoints();
    brownianPoints.SetInputConnection(sphereSource.GetOutputPort());

    vtkArrowSource arrowSource = new vtkArrowSource();

    vtkGlyph3D glyph3D = new vtkGlyph3D();
    glyph3D.SetSourceConnection(arrowSource.GetOutputPort());
    glyph3D.SetInputConnection(brownianPoints.GetOutputPort());
    glyph3D.SetScaleFactor(.3);
    
    //Create a mapper and actor for sphere
    vtkPolyDataMapper mapper = new vtkPolyDataMapper();
    mapper.SetInputConnection(sphereSource.GetOutputPort());

    vtkActor actor = new vtkActor();
    actor.GetProperty().EdgeVisibilityOn();
    actor.GetProperty().SetInterpolationToFlat();
    actor.SetMapper(mapper);

    //Create a mapper and actor for glyphs
    vtkPolyDataMapper glyphMapper = new vtkPolyDataMapper();
    glyphMapper.SetInputConnection(glyph3D.GetOutputPort());

    vtkActor glyphActor = new vtkActor();
    glyphActor.GetProperty().SetColor(glyphActorColor);
    glyphActor.SetMapper(glyphMapper);
                
    //Create the renderer, render window and interactor.
    vtkRenderer ren = new vtkRenderer();
    vtkRenderWindow renWin = new vtkRenderWindow();
    renWin.AddRenderer(ren);
    vtkRenderWindowInteractor iren = new vtkRenderWindowInteractor();
    iren.SetRenderWindow(renWin);
    
    ren.AddActor(actor);
    ren.AddActor(glyphActor);
    
    ren.ResetCamera();
    ren.GetActiveCamera().Azimuth(30	);
    ren.GetActiveCamera().Elevation(30);
    ren.GetActiveCamera().Dolly(1.4);
    ren.ResetCameraClippingRange();
    ren.SetBackground(BgColor);
    
    renWin.SetSize(300,300);
    renWin.Render();

    iren.Initialize();
    iren.Start();

  }
}
